Integrate netatalk 2.x branch

NJRoadfan opened the pull request:
Integrate rdmark's netatalk-2.x branch into A2SERVER.
Update A2SERVER scripts to support systemd.

NJRoadfan committed:
Resync with master

NJRoadfan committed:
Go bleeding edge (for now)

NJRoadfan committed:
Convert netatalk install to systemd

NJRoadfan committed:
Add macipgw startup script

NJRoadfan committed:
Add macipgw systemd service

NJRoadfan committed:
More legacy config removal

rdmark committed:
New env variable that controls whether to copy local script files or not

rdmark committed:
Remove ancient firmware workaround that bricks modern RPis

rdmark committed:
Option to compile multi threaded

rdmark committed:
Multi threaded compilation for nulib2

NJRoadfan committed:
Merge pull request #7 from rdmark/rdmark-remove-firmware-fix

Remove ancient firmware workaround that bricks modern RPis

NJRoadfan committed:
Merge pull request #6 from rdmark/rdmark-hacks

Introduce options to use local script files, and to compile multi thread

rdmark committed:
Bump to 2.230202 release version. Download bootstrapped tarball and skip bootstrap step. Allow reinstalling Netatalk from source.

NJRoadfan committed:
Merge pull request #8 from rdmark/rdmark-netatalk-check

Use Netatalk 2.230202 release tag; allow for reinstalling Netatalk from source

rdmark committed:
Streamline Netatalk dep packages to install; remove backwards compat code (#9)

* Streamline Netatalk dep packages to install; remove old Debian distro support; remove pre-compiled package code.

NJRoadfan committed:
Update a2server-aliases.txt

Update aliases for systemd configuration.

NJRoadfan committed:
Remove some legacy cruft and more systemd integration

rdmark committed:
Bump to netatalk 2.230301 (#10)

NJRoadfan committed:
Update macipgw.service.txt

NJRoadfan committed:
Bump up to latest netatalk-2.x release

NJRoadfan committed:
Bump version number

NJRoadfan committed:
Merge branch 'njroadfan-main-dev' into new-netatalk-test

NJRoadfan committed:
Explicitly install cups-browsed

Using "--no-install-recommends" was not installing this package automatically. It is needed for driverless (AirPrint, etc) printer discovery. Without it, CUPS and thus papd, can't automatically detect and add driverless printer queues.

NJRoadfan committed:
Update a2server-aliases.txt

Fix typo

NJRoadfan committed:
Restart CUPS the systemd way

NJRoadfan committed:
AppleTalk is always on

So remove the option. Doesn't work with systemd anymore anyway.

NJRoadfan committed:
Remove obsolete options from help.

NJRoadfan committed:
Make sure CNID is enabled/disabled when using alias

NJRoadfan committed:
Make sure we start/stop CNID

NJRoadfan committed:
Make sure avahi-deamon is installed with CUPS

NJRoadfan committed:
Always start netatalk services in the background

rdmark commented:
Heads-up that I tagged a new release earlier today: https://github.com/rdmark/netatalk-2.x/releases/tag/netatalk-2-230701

NJRoadfan committed:
Update to netatalk 2.x 230701

NJRoadfan committed:
Set correct interface name for RPi

Raspberry Pi OS is still using eth0 for the built-in Ethernet interface by default.

NJRoadfan committed:
Revert "Update URLs"

This reverts commit 5d6f689b87e0251963d69556e8932fe887a3f573.

NJRoadfan committed:
Update for Bookworm

Deprecate older releases of Debian since they are pretty ancient and unsupported.

NJRoadfan committed:
Allow guests to read/write

Guest account needs the same permissions as a user, otherwise we can't write files in ProDOS 8.

NJRoadfan committed:
Switch to mainline netatalk 2.2.10

The unofficial netatalk-2.x branch has been discontinued. Switch to the official repository.

NJRoadfan committed:
Update for bookworm

NJRoadfan committed:
Update supported OSes

NJRoadfan committed:
Remove RPi version checks

Obsolete code at this point.

NJRoadfan committed:
Update macipgw-start.sh.txt

Make sure we select an ethernet interface and not a tap interface.

NJRoadfan committed:
Update to netatalk 2.3.2

NJRoadfan committed:
Update a2server-3-sharing.txt

Typo in the tar command.

NJRoadfan committed:
Add macip-prefs.tgz to repo

Instead of relying on Ivan's server.

NJRoadfan committed:
Update a2server-5-netboot.txt Change macip-prefs.tgz path

NJRoadfan committed:
Update a2server-3-sharing.txt disable netatalk's home folder name

NJRoadfan committed:
Update a2server-3-sharing.txt Fix sed command in previous commit

NJRoadfan committed:
Update a2server-5-netboot.txt - Add NetDisk 1.0

Add Stephen Heumann's NetDisk 1.0 and the Internet Archive browser.

NJRoadfan committed:
Update a2server-5-netboot.txt

NJRoadfan committed:
Update a2server-5-netboot.txt Add AFPBridge1.0b1

Add Stephen Heumann's AFPBridge to allow use of TCP/IP only AFP servers.

NJRoadfan committed:
Update a2server-5-netboot.txt Fix typo

NJRoadfan committed:
Update a2server-5-netboot.txt Install 8-bit AppleShare Tools

Install Chooser II, Namer, and other tools for 8-bit clients.

NJRoadfan committed:
Update a2server-5-netboot.txt

NJRoadfan committed:
Update a2server-5-netboot.txt

NJRoadfan committed:
Update a2server-5-netboot.txt Make sure user's FinderExtras Folder exists

NJRoadfan committed:
Update a2server-5-netboot.txt Set proper filetypes for ATINIT files

Fixes a longstanding bug. While GS/OS and ProDOS can read ATINIT files without the proper file type and allow netbooting, other programs will not work correctly. For example, one wouldn't be able to save their printer information in GS/OS or use the 8-bit Chooser II program.

NJRoadfan committed:
Update a2server-5-netboot.txt

Try this again

NJRoadfan committed:
Update a2server-5-netboot.txt Quiet things down.

rdmark commented on scripts/tools/a2server-aliases.txt:
```diff
@@ -17,17 +17,16 @@ alias rasppleii-update='raspbian-update a2cloud a2server'
+alias showip-wifi='ip a show dev wlan0 | grep "inet " | cut -dt -f2 | cut -db -f1'
+alias showmac-wifi='ip a show dev wlan0 | grep "link/ether" | cut -dr -f2 | cut -db -f1'
+
+alias netatalk-stop="sudo systemctl stop atalkd cnid afpd papd timelord a2boot macipgw; bonjour-off"
+alias netatalk-start="sudo systemctl start atalkd cnid afpd papd timelord a2boot macipgw &> /dev/null; bonjour-on"
+alias netatalk-restart="sudo systemctl restart atalkd cnid afpd papd timelord a2boot macipgw &> /dev/null; bonjour-on"
```
The `cnid` service is designed to be started on demand by the `afpd` process. It should not be controlled directly.

NJRoadfan committed:
Install ImageWriter drivers and Namer for GS/OS

NJRoadfan committed:
Don't restart CNID deamon directly

The systemd script for afpd controls this.

NJRoadfan committed:
Update a2server-aliases.txt Don't directly control CNID service.

NJRoadfan committed:
Update a2server-5-netboot.txt Update P8 Logon

Update the Logon program to v1.5 to match boot blocks.

NJRoadfan committed:
Update a2server-5-netboot.txt Check for missing P8 Tools

NJRoadfan committed:
Update a2server-6-samba.txt Finally fix this for systemd

Still has the lame init.d file detection, but Samba still seems to install it even on systemd installs.

NJRoadfan committed:
Update a2server-3-sharing.txt

NJRoadfan committed:
Update a2server-3-sharing.txt Stop restarting CUPS

For some reason CUPS hangs when restarted at this point.

NJRoadfan committed:
Update a2server-aliases.txt Update Samba control for systemd

NJRoadfan committed:
Update a2server-3-sharing.txt Update systemd service path

NJRoadfan committed:
Grab P8 AppleTalk Tools from Ivan's server instead.

NJRoadfan committed:
Update mkatinit to set ProDOS file types.

Move this out of the netboot script.

NJRoadfan committed:
Cleanup ATINIT file creation

We moved this to the mkainit script.

NJRoadfan committed:
New Feature: Add TashRouter to A2SERVER (#13)

* Create tashrouter.service.txt

* Create tashrouter.py

This sets up and executes TashRouter. Heavily based off Tashtari's example, but with added support for gracefully terminating via SIGTERM like a real service should do.

* Create tash0.txt

This goes into /etc/network/interfaces.d. We need this to create persistent TAP interface on bootup to link TashRouter and netatalk's atalkd together.

* We need to install uml-utilities

* Update a2server-3-sharing.txt

* Remove Netatalk Router on/off alias

We can't risk the user bricking their setup now that atalkd is managing two interfaces.

NJRoadfan committed:
Update to Netatalk 2.4.0 and use meson to build

NJRoadfan committed:
Update a2server-3-sharing.txt

NJRoadfan committed:
Update a2server-3-sharing.txt

NJRoadfan committed:
Update a2server-3-sharing.txt

NJRoadfan committed:
Update Ewan's tools to the latest revisions

NJRoadfan committed:
Update a2server-5-netboot.txt

NJRoadfan committed:
Update a2server-5-netboot.txt

NJRoadfan committed:
Only create a TAP interface if needed.

NJRoadfan committed:
Add TashTalk Hat Support to A2SERVER (#14)

* Update index.txt Add TashTalk switches

* Update a2server-3-sharing.txt Add TashTalk Hat Install

NJRoadfan committed:
Update a2server-3-sharing.txt Misc updates

NJRoadfan committed:
Update to Netatalk 2.4.1

NJRoadfan committed:
Update a2server-3-sharing.txt

NJRoadfan committed:
Figures that Debian releases an update right after I changed this.
